src/types/assertion.rs

```diff
@@ -0,0 +1,661 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub struct Assertion {
+    root: Op,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(tag = "cmp")]
+#[serde(rename_all = "snake_case")]
```

Comment:
I froget, can you combine these
```suggestion
#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
#[serde(tag = "cmp", rename_all = "snake_case")]
```